### ISSUE TITLE
Autologin option

### ISSUE THREAD

**Issue by rumkin** [MrSwitch/hello.js#124]
Hello.js store login data in local storage and automatically login on init.  Add an option not to autologin on init.

**Comment by cordphelps** [comment]
guys, if autologin is triggered, should the client_ID be matched to indicate success (?)

I'm confused about the case where one browser tab has max privileges/scope (gmail login), and I'm running my app in another tab expecting limited privileges/scope. The app autologin token seems to have max privileges.

**Comment by MrSwitch** [comment]
@cordphelps The tokens are stored in localStorage which transcend browser windows and tabs.

If you want to distinguish different environments you could copy and rename the services,

i.e.
```javascript
hello.services.twitter2 = hello.services.twitter;
hello.init({
   'twitter' : id
   'twitter2' : id2 // but could be the same
});
hello('twitter2').login();
... etc.
```
Makes sense?

**Comment by cordphelps** [comment]
sorry, not quite.

in the case where the user has already authenticated himself and authorized a scope in another tab (for gmail, for instance); I launch my app and hello.js triggers the auth.login event and picks up the existing token (from localstorage) via 'autologin' behavior;

that token represents the user I care about but does not authorize the limited scope that my app intends. What do I do?

thank you!

**Comment by MrSwitch** [comment]
If the user has granted max privileges initially for the app. The second tab is going to have access to all of them if it shares the same localStorage, I.e. has the on the same origin.
I don't understand the problem this creates. Perhaps you can explain how this impacts your user.

**Comment by cordphelps** [comment]
maybe I am overthinking this, I appreciate your feedback. I put my sandbox
out there:

https://github.com/cordphelps/tokenCheck

**Comment by philsturgeon** [comment]
Hey there! This is a bit of a necrobump for sure, but I've got a use case for you. 

We have a React app, trying to log users in via Google. We authenticate with Google using hello, then on success we go off to a third party system to see if those Google creds are good enough to let them into the admin panel.

If they are not allowed in (using my personal google account instead of the work one), every time they load the login page, due to having `hello` in local storage, hellos `auth.login` event will be triggered. This leads to the success event happening, but  a "Login failed" error to show up. 

This sucks for us, because they are not trying to log in, just loading the login page. Before even doing anything they're being told they did it wrong.

If I try to remove `hello` from local storage it triggers `auth.logout`, which has other side effects.

Basically I want hello to stop automagically trying to do stuff and let me tell it when to do stuff.

I'll implement this if you can give me advice on the approach. I figure just a boolean option that gets added to these conditions?

```js
			// Access_token has been removed
			else if (!session.access_token && oldSess.access_token) {
				emit('logout');
			}

			// Access_token has been created
			else if (session.access_token && !oldSess.access_token) {
				emit('login');
			}
```

**Comment by philsturgeon** [comment]
I had a really quick stab at this, and it's solved my use case.

https://github.com/MrSwitch/hello.js/pull/449

Basically, I only want hello to do stuff when I tell it to. If this is not the best approach then let me know, but the following is really easy to work with:

```
    hello.init({
      google: Config.google.id
    }, {
      autoemit: false,
      ...
    });
```

**Comment by MrSwitch** [comment]
@philsturgeon Could you not bind to the login handler `hello(path).login().then(handleLogin)` ? 

Or put your autoemit flag in global events `hello.on('login', () => if (config.autoemit) {...});` ?

**Comment by philsturgeon** [comment]
Hey @MrSwitch, I think I must not be explaining the problem or showing the solution very well. Would you take a look at https://github.com/MrSwitch/hello.js/pull/449 and maybe continue the chat over there?

**Comment by andrejsc** [comment]
I need this too.
